pkg/clusterapi/identity.go

```diff
@@ -103,19 +103,34 @@ func configureOIDCInKubeadmControlPlane(kcp *controlplanev1.KubeadmControlPlane,
 	}
 }
 
+func configureAPIServerExtraArgsInKubeadmControlPlane(kcp *controlplanev1.KubeadmControlPlane, apiServerExtraArgs map[string]string) {
+	if apiServerExtraArgs == nil {
+		return
+	}
+
+	for k, v := range apiServerExtraArgs {
+		kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.APIServer.ExtraArgs[k] = v
+	}
+}
+
 func configurePodIamAuthInKubeadmControlPlane(kcp *controlplanev1.KubeadmControlPlane, podIamConfig *v1alpha1.PodIAMConfig) {
 	if podIamConfig == nil {
 		return
 	}
 
 	apiServerExtraArgs := kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.APIServer.ExtraArgs
-	for k, v := range PodIAMAuthExtraArgs(podIamConfig) {
-		apiServerExtraArgs[k] = v
+	if _, ok := apiServerExtraArgs["service-account-issuer"]; ok && podIamConfig.ServiceAccountIssuer != "" {
+		apiServerExtraArgs["service-account-issuer"] += "," + podIamConfig.ServiceAccountIssuer
```

Comment:
I see the concern here, this seems like a mix of assuming there is only one field and then a for loop below right? I think we should try to be consistent, either remove the for loop below or do this check within that for loop where if service-account-issuer already exists, then append.

Reply:
There is a for loop here as well. It's hidden within the `Append()` function

Reply:
Added the condition within the for-loop for all providers